firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -181,6 +225,29 @@ public void onResumed(Activity activity) {
     }
   }
 
+  @Nullable
+  private static <A extends Activity> Activity getActivityWithIgnoredClass(
+      Activity activity, @Nullable Activity fallbackActivity, @Nullable Class<A> classToIgnore) {
+    if (classToIgnore != null && classToIgnore.isInstance(activity)) {
+      return fallbackActivity;
+    } else {
+      return activity;
+    }
+  }
+
+  private void updateCurrentActivity(@Nullable Activity activity) {
+    synchronized (lock) {
+      if (currentActivity != activity) {
+        if (currentActivity != null) {
+          // Store a reference to the previous activity in case the current activity is ignored
+          // later in call to applyToNullableForegroundActivity()
+          previousActivity = currentActivity;
```

Comment:
I'm not sure if it's actually a problem in practice in you use case, but in general holding a reference to `currentActivity` should be safe, while it's not necessarily the case with `previousActivity` given that that activity may get destroyed while `currentActivity` is active. Is that a possible scenario here? What would happen if you try to use that activity?

Reply:
That's not a possible scenario because if `previousActivity` gets destroyed, we remove the reference to it on line 343 (in `onActivityDestroyed`).